runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -0,0 +1,1529 @@
+				result = vmFuncs->j9jni_createLocalRef(env, clazzObject);
+			} else {
+				vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, NULL);
+			}
+		}
+	}
+	Trc_JCL_java_lang_invoke_MethodHandleNatives_staticFieldBase_Exit(env, result);
+	vmFuncs->internalExitVMToJNI(currentThread);
+	return result;
+}
+
+/**
+ * static native Object getMemberVMInfo(MemberName self);  // returns {vmindex,vmtarget}
+ * 
+ * Return a 2-element java array containing the vm offset/target data
+ * For a field MemberName, array contains:
+ * 		(field offset, declaring class)
+ * For a method MemberName, array contains:
+ * 		(vtable index, MemberName object)
+ */
+jobject JNICALL
+Java_java_lang_invoke_MethodHandleNatives_getMemberVMInfo(JNIEnv *env, jclass clazz, jobject self)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	const J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	jobject result = NULL;
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+
+	Trc_JCL_java_lang_invoke_MethodHandleNatives_getMemberVMInfo_Entry(env, self);
+	if (NULL != self) {
+		J9Class *arrayClass = fetchArrayClass(currentThread, J9VMJAVALANGOBJECT(vm));
+		j9object_t arrayObject = vm->memoryManagerFunctions->J9AllocateIndexableObject(currentThread, arrayClass, 2, J9_GC_ALLOCATE_OBJECT_INSTRUMENTABLE);
+		if (NULL == arrayObject) {
+			vmFuncs->setHeapOutOfMemoryError(currentThread);
+		} else {
+			PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, arrayObject);
+			j9object_t box = vm->memoryManagerFunctions->J9AllocateObject(currentThread, J9VMJAVALANGLONG_OR_NULL(vm), J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+			if (NULL == box) {
+				/* Drop arrayObject */
+				DROP_OBJECT_IN_SPECIAL_FRAME(currentThread);
+				vmFuncs->setHeapOutOfMemoryError(currentThread);
+			} else {
+				arrayObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
+				j9object_t membernameObject = J9_JNI_UNWRAP_REFERENCE(self);
+				jint flags = J9VMJAVALANGINVOKEMEMBERNAME_FLAGS(currentThread, membernameObject);
+				jlong vmindex = (jlong)(UDATA)J9OBJECT_U64_LOAD(currentThread, membernameObject, vm->vmindexOffset);
+				j9object_t target;
+				if (J9_ARE_ANY_BITS_SET(flags, MN_IS_FIELD)) {
+					vmindex = ((J9JNIFieldID*)vmindex)->offset;
+					target = J9VMJAVALANGINVOKEMEMBERNAME_CLAZZ(currentThread, membernameObject);
+				} else {
+					J9JNIMethodID *methodID = (J9JNIMethodID*)vmindex;
+					if (J9_ARE_ANY_BITS_SET(methodID->vTableIndex, J9_JNI_MID_INTERFACE)) {
+						vmindex = methodID->vTableIndex & ~J9_JNI_MID_INTERFACE;
+					} else if (0 == methodID->vTableIndex) {
+						vmindex = -1;
+					} else {
+						vmindex = methodID->vTableIndex;
+					}
+					target = membernameObject;
+				}
+
+				J9VMJAVALANGLONG_SET_VALUE(currentThread, box, vmindex);
+				J9JAVAARRAYOFOBJECT_STORE(currentThread, arrayObject, 0, box);
+				J9JAVAARRAYOFOBJECT_STORE(currentThread, arrayObject, 0, target);
+
```

Comment:
Why are both of these writing to index 0?

Reply:
fixed